update docs for beta_inc, gamma, gamma_inc_inv, zeta, bessels

AshtonSBradley opened the pull request:
This PR only changes docstrings. Have added `See also` and `External links` where I can see an obvious way to do it.

- `beta_inc`: changed function signature to `beta_inc(a,b,x)`, and made explanation consistent. This is simpler to explain and call since `y=1-x`, is enforced in the `beta_inc(a,b,x,y)` form anyway.  
- Tidy ups in `gamma`, `gamma_inc`, `gamm_inc_inv`, `zeta`. 
- bessel functions: cross and external links

simonbyrne commented:
Looks great, thanks!

DLMF is also a good place to add links to, since it is fairly authoritative

AshtonSBradley commented:
ok, I will go through and add DLMF links for these updates too. I also wasn't sure whether to be linking to e.g. precise subsection on wikipedia, or just the Bessel Function page which is presumably less changeable. Do you have any advice on this?

AshtonSBradley commented:
- Added cross and external links for `airy`, `bessel`, `hankel`. I have linked to specific Wikipedia subsections where appropriate.
- Added some more DLMF links
- Fixed a typo in `gammax` docstring
- Few more minor tidy ups

simonbyrne commented on src/bessel.jl:
```diff
@@ -594,26 +681,42 @@ end
 
 Bessel function of the third kind of order `nu`, ``H^{(1)}_\\nu(x)``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Hankel_functions:_H(1)%CE%B1,_H(2)%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.5), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Hankel_functions:_H(1)%CE%B1,_H(2)%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -594,26 +681,42 @@ end
 
 Bessel function of the third kind of order `nu`, ``H^{(2)}_\\nu(x)``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Hankel_functions:_H(1)%CE%B1,_H(2)%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.6), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Hankel_functions:_H(1)%CE%B1,_H(2)%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -594,26 +681,42 @@ end
 
 Scaled Bessel function of the third kind of order `nu`, ``H^{(1)}_\\nu(x) e^{-x i}``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Hankel_functions:_H(1)%CE%B1,_H(2)%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.5), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Hankel_functions:_H(1)%CE%B1,_H(2)%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -594,26 +681,42 @@ end
 
 Scaled Bessel function of the third kind of order `nu`, ``H^{(2)}_\\nu(x) e^{x i}``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Hankel_functions:_H(1)%CE%B1,_H(2)%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.6), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Hankel_functions:_H(1)%CE%B1,_H(2)%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -415,6 +462,10 @@ end
 
 Bessel function of the first kind of order `nu`, ``J_\\nu(x)``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_first_kind:_J%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_first_kind:_J%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -457,6 +516,10 @@ end
 
 Scaled modified Bessel function of the second kind of order `nu`, ``K_\\nu(x) e^x``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.25), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Modified_Bessel_functions:_I%CE%B1,_K%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.25.3), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Modified_Bessel_functions:_I%CE%B1,_K%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -471,6 +534,10 @@ end
 
 Bessel function of the second kind of order `nu`, ``Y_\\nu(x)``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_second_kind:_Y%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.3), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_second_kind:_Y%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -486,6 +553,10 @@ end
 Scaled Bessel function of the second kind of order `nu`,
 ``Y_\\nu(x) e^{- | \\operatorname{Im}(x) |}``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_second_kind:_Y%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.3), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_second_kind:_Y%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -529,6 +600,10 @@ end
 
 Bessel function of the first kind of order 0, ``J_0(x)``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_first_kind:_J%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_first_kind:_J%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -540,6 +615,10 @@ end
 
 Bessel function of the first kind of order 1, ``J_1(x)``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_first_kind:_J%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_first_kind:_J%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -571,6 +654,10 @@ end
 
 Bessel function of the second kind of order 1, ``Y_1(x)``.
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_second_kind:_Y%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.3), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Bessel_functions_of_the_second_kind:_Y%CE%B1)
```

simonbyrne commented on src/bessel.jl:
```diff
@@ -267,7 +299,10 @@ end
 `k` defaults to 1 if it is omitted.
-(See also [`besselhx`](@ref) for an exponentially scaled variant.)
+
+External links: [DLMF](https://dlmf.nist.gov/10.2), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Hankel_functions:_H(1)%CE%B1,_H(2)%CE%B1)
```
```suggestion
External links: [DLMF](https://dlmf.nist.gov/10.2.5) and [DLMF](https://dlmf.nist.gov/10.2.6), [Wikipedia](https://en.wikipedia.org/wiki/Bessel_function#Hankel_functions:_H(1)%CE%B1,_H(2)%CE%B1)
```

AshtonSBradley commented:
I don't understand the error 
`linux - arm - Julia 1.3: Error: No such image: docker.io/library/julia:1.3`
Is there something I should fix?

giordano commented:
>  Is there something I should fix?

No, there is still no official build for Julia 1.3.1 on ARM

simonbyrne commented:
Thanks!
